encrypt_util.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021 Snowflake Computing Inc. All right reserved.
+
+package gosnowflake
+
+//lint:file-ignore U1000 TODO SNOW-29352
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+)
+
+type snowflakeFileEncryption struct {
+	QueryStageMasterKey string `json:"queryStageMasterKey,omitempty"`
+	QueryID             string `json:"queryId,omitempty"`
+	SMKID               int64  `json:"smkId,omitempty"`
+}
+
+// PUT requests return a single encryptionMaterial object whereas GET requests
+// return a slice (array) of encryptionMaterial objects, both under the field
+// 'encryptionMaterial'
+type encryptionWrapper struct {
+	snowflakeFileEncryption
+	EncryptionMaterials []snowflakeFileEncryption
+}
+
+// override default behavior for wrapper
+func (ew *encryptionWrapper) UnmarshalJSON(data []byte) error {
+	// if GET, unmarshal slice of encryptionMaterial
+	if err := json.Unmarshal(data, &ew.EncryptionMaterials); err == nil {
+		return err
+	}
+	// else (if PUT), unmarshal the encryptionMaterial itself
+	return json.Unmarshal(data, &ew.snowflakeFileEncryption)
+}
+
+type encryptMetadata struct {
+	key     string
+	iv      string
+	matdesc string
+}
+
+func encryptStream(sfe *snowflakeFileEncryption, src *bytes.Buffer, out *bytes.Buffer, chunkSize int) *encryptMetadata {
```

Comment:
A small doc string explaining what this functions is employing would be nice. Something on the lines of 
`... this function encrypts a stream buffer using AES128 block cipher in CBC mode with PKCS5 padding.`